lib/perl/centreon/script/centcore.pm

```diff
@@ -467,194 +465,144 @@ sub removeIllegalCharacters($) {
-                    && $server_info->{remote_server_centcore_ssh_proxy} == 1
-                ) {
-                    $cmd_line .= 'EXTERNALCMD:' . $id . ':' . $self->removeIllegalCharacters($cmd1) . "\n";
-                } else {
-                    $cmd_line .= $self->removeIllegalCharacters($cmd1) . "\n";
-                }
+            foreach my $command (@{$commandArray}) {
+                $command =~ s/\'/\'\\\'\'/g;
+                $command = $self->removeIllegalCharacters($command);
                 $count++;
                 $totalCount++;
 
-                if ($count >= 200 || $totalCount == $countCommands) {
-                    if (defined($server_info->{remote_id}) 
-                        && $server_info->{remote_id} != 0
-                        && $self->{instance_mode} ne "remote"
-                        && $server_info->{remote_server_centcore_ssh_proxy} == 1
-                    ) {
-                        # Forward commands to Remote Server Master
-                        my $remote_server = $self->getServerConfig($server_info->{remote_id});
-                        $port = checkSSHPort($remote_server->{ssh_port});
-                        $cmd_line =~ s/^\s+|\s+$//g;
-                        $cmd2 = "$self->{ssh} -q " . $remote_server->{ns_ip_address} . " -p $port "
-                            . "\"$self->{echo} '" . $cmd_line . "' >> " . $self->{cmdDir} . time() . "-sendcmd\"";
-                        $self->{logger}->writeLogInfo(
-                            "Sending external command using Remote Server: " . $remote_server->{name}
-                        );
-
-                        ($lerror, $stdout, $return_code) = centreon::common::misc::backtick(
-                            command => $cmd2,
-                            logger => $self->{logger},
-                            timeout => $self->{cmd_timeout},
-                            wait_exit => 1
-                        );
-                        
-                        if ($lerror != 0 || (defined($return_code)  && $return_code != 0)) {
-                            $self->{logger}->writeLogError(sprintf(
-                                "Couldn't send external command %s on Remote Server: %s (%d)", 
-                                $cmd_line,
-                                $remote_server->{name},
-                                $remote_server->{id}
-                                )
-                            );
+                if (defined($serverInfo->{remote_id}) && $serverInfo->{remote_id} != 0
+                    && $self->{instance_mode} ne "remote" && $serverInfo->{remote_server_centcore_ssh_proxy} == 1) {
+                    # Add Centcore MACRO when routing to Remote Server
+                    $command = "EXTERNALCMD:" . $pollerId . ":" . $command;
+                    $multipleCommands .= $command . "\n";
+                } else {
+                    $multipleCommands .= $command . "\n";
```

Comment:
this line could be put outside of the instruction, avoiding to put it in "if" and "else"

Reply:
Indeed.